subiquity/models/mirror.py

```diff
@@ -13,35 +13,61 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import copy
 import logging
-import platform
 from urllib import parse
 
+from curtin.commands.apt_config import (
+    get_arch_mirrorconfig,
+    get_mirror,
+    PRIMARY_ARCHES,
+    )
+from curtin.util import get_architecture
+
 log = logging.getLogger('subiquitycore.models.mirror')
 
-# correct default mirror for most arches
-DEFAULT_MIRROR = 'http://ports.ubuntu.com/ubuntu-ports'
-# apart from the two snowflakes
-if platform.machine() in ['i686', 'x86_64']:
-    DEFAULT_MIRROR = 'http://archive.ubuntu.com/ubuntu'
+
+DEFAULT = {
+    "preserve_sources_list": False,
+    "primary": [
+        {
+            "arches": PRIMARY_ARCHES,
+            "uri": "http://archive.ubuntu.com/ubuntu",
+        },
+        {
+            "arches": ["default"],
+            "uri": "http://ports.ubuntu.com/ubuntu-ports",
+        },
+        ],
+}
+
+
+def get_default_mirror():
+    return get_mirror(DEFAULT, "primary", get_architecture())
```

Comment:
Could we move the get_mirror() call to the object initialization so we can cache it?  It won't change during the lifetime of the install so we can save some execs and overhead if we keep it around.

Reply:
Alternatively, we recently made use of python3's lru_cache() in cloud-init

from functools import lru_cache()

@lru_cache()
def get_default_mirror()

...

Reply:
Tweaked a bit, see what you think now.